src/_P105_AHT.ino

```diff
@@ -0,0 +1,206 @@
+# define PLUGIN_ID_105         105
+# define PLUGIN_NAME_105       "Environment - AHT10/20/21 [TESTING]"
+# define PLUGIN_VALUENAME1_105 "Temperature"
+# define PLUGIN_VALUENAME2_105 "Humidity"
+
+
+boolean Plugin_105(uint8_t function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+      Device[++deviceCount].Number           = PLUGIN_ID_105;
+      Device[deviceCount].Type               = DEVICE_TYPE_I2C;
+      Device[deviceCount].VType              = Sensor_VType::SENSOR_TYPE_TEMP_HUM;
+      Device[deviceCount].Ports              = 0;
+      Device[deviceCount].PullUpOption       = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption      = true;
+      Device[deviceCount].ValueCount         = 2;
+      Device[deviceCount].SendDataOption     = true;
+      Device[deviceCount].TimerOption        = true;
+      Device[deviceCount].GlobalSyncOption   = true;
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_105);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_105));
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_105));
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SHOW_I2C_PARAMS:
+    {
+      int Plugin_28_i2c_addresses[2] = { 0x38, 0x39 };
+      addFormSelectorI2C(F("i2c_addr"), 2, Plugin_28_i2c_addresses, PCONFIG(0));
+      addFormNote(F("SDO Low=0x38, High=0x39. NB: Only available on AHT10 sensors."));
+
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      if (static_cast<AHTx_device_type>(PCONFIG(1)) ==  AHT10_DEVICE) {
+        bool hasOtherI2CDevices = false;
+
+        for (taskIndex_t x = 0; validTaskIndex(x) && !hasOtherI2CDevices; x++) {
+          const deviceIndex_t DeviceIndex = getDeviceIndex_from_TaskIndex(x);
+
+          if (validDeviceIndex(DeviceIndex)
+              && (Settings.TaskDeviceDataFeed[x] == 0)
+              && ((Device[DeviceIndex].Type == DEVICE_TYPE_I2C)
+                # ifdef PLUGIN_USES_SERIAL // Has I2C Serial option
+                  || (Device[DeviceIndex].Type == DEVICE_TYPE_SERIAL)
+                  || (Device[DeviceIndex].Type == DEVICE_TYPE_SERIAL_PLUS1)
+                # endif // ifdef PLUGIN_USES_SERIAL
+                  )
+              ) {
+            hasOtherI2CDevices = true;
+          }
+        }
+
+        if (hasOtherI2CDevices) {
+          addRowLabel(EMPTY_STRING, EMPTY_STRING);
+          addHtmlDiv(F("note warning"),
+                     F("Attention: Sensor model AHT10 may cause I2C issues when combined with other I2C devices on the same bus!"));
+        }
+      }
+      {
+        const String options[] = { F("AHT10"), F("AHT20"), F("AHT21") };
+        int indices[]          = { AHT10_DEVICE, AHT20_DEVICE, AHT21_DEVICE };
```

Comment:
Just nitpicking, but these indices can also be defined as `const`.
Maybe the compiler then can optimize it slightly more.

Reply:
I'll use that more often then: no need to change -> make `const` 😃